mat/band_test.go

```diff
@@ -0,0 +1,144 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mat
+
+import (
+	"reflect"
+	"testing"
+
+	"gonum.org/v1/gonum/blas/blas64"
+)
+
+func TestNewBand(t *testing.T) {
+	for i, test := range []struct {
+		data   []float64
+		r, c   int
+		kl, ku int
+		mat    *BandDense
+	}{
+		{
+			data: []float64{
+				-1, 1, 2, 3,
+				4, 5, 6, 7,
+				8, 9, 10, 11,
+				12, 13, 14, 15,
+				16, 17, 18, -1,
+				19, 20, -1, -1,
+			},
+			r: 6, c: 6,
+			kl: 1, ku: 2,
+			mat: &BandDense{
+				mat: blas64.Band{
+					Rows:   6,
+					Cols:   6,
+					KL:     1,
+					KU:     2,
+					Stride: 4,
+					Data: []float64{
+						-1, 1, 2, 3,
+						4, 5, 6, 7,
+						8, 9, 10, 11,
+						12, 13, 14, 15,
+						16, 17, 18, -1,
+						19, 20, -1, -1,
+					},
+				},
+			},
+		},
+	} {
+		band := NewBandDense(test.r, test.c, test.kl, test.ku, test.data)
+		rows, cols := band.Dims()
+
+		if rows != test.r {
+			t.Errorf("unexpected number of rows for test %d: got: %d want: %d", i, rows, test.r)
+		}
+		if cols != test.c {
+			t.Errorf("unexpected number of cols for test %d: got: %d want: %d", i, cols, test.c)
+		}
+		if !reflect.DeepEqual(band, test.mat) {
+			t.Errorf("unexpected data slice for test %d: got: %v want: %v", i, band, test.mat)
+		}
+	}
+}
+
+func TestBandAtSet(t *testing.T) {
+	band := NewBandDense(6, 6, 1, 2, []float64{
```

Comment:
It would be nice to have a test that just accesses all of the matrix elements to make sure the index tests are correct.